vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueType
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
```

Comment:
Off-the-cuff without knowing anything about JSI (applies to all my comments on this change): Does this enum need an 'Array' member?

Reply:
No, these are JavaScript basic value types where Array is just an Object.

---
In reply to: [451842892](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451842892) [](ancestors = 451842892)

Reply:
@ : I ran into this question too. Note that the same is true for functions. in JSI you can ask an object if they are actually an array or function...

---
In reply to: [454515553](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r454515553) [](ancestors = 454515553,451842892)